### ISSUE TITLE
Add label to mobile menu dropdown button

### ISSUE THREAD

**Issue by HaSistrunk** [RockefellerArchiveCenter/dimes#848]
## Describe the bug
Accessibility audit issue 102: On small screens, Nav menu dropdown button in the header does not have a text alternative, which is confusing for screen reader users and a potential blocker for voice activation. This is a violation of WCAG 4.1.2 Name, role, value, and is a potential blocker for screen reader users.

## To reproduce
1. With a screen reader, navigate to the mobile menu dropdown button. 
2. Hear that it is not announced

## Additional context
Solutions include either adding a visually hidden text label or aria-label to the button element, removing the aria-hidden attrribute on the span element so that the name would be "menu".
